### ISSUE TITLE
Optional chaining arg default with this

### ISSUE THREAD

**Issue by samdenty** [microsoft/TypeScript#37852]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:** Optional chaining, argument default this

**Code**

```ts
class Test {
    public c(a = this.a?.b) {}
}
```

**Expected behavior:**

Shouldn't throw reference error

**Actual behavior:**

Throws reference error for _a
```js
c(a = (_a = this.a) === null || _a === void 0 ? void 0 : _a.b) {
  var _a;
}
```

**Playground Link:** https://www.typescriptlang.org/play/index.html#code/FAYwNghgzlAEAqBTKAXWBvYtuwA4FcAjMASxFglgF4NZCAuWARlgF9gOc8jTyQAKSjRQALElAB0EAPwTCASgycc7dsAB2iAO4JkKfvIkD5QA

**Comment by DanielRosenwasser** [comment]
Duplicate of #36295
